GOVERNANCE.md

```diff
@@ -0,0 +1,251 @@
+    is
+    * Subject line not longer than 50 characters
+    * Subject capitalized
+    * No period at the end of the subject
+    * Imperative mood in the subject
+    * Second line empty
+    * Body lines not longer than 72 characters
+
+  Commits using the `--fixup` or `--squash` feature of `git-commit` are allowed,
+  but won't be merged (read more about this in the "merge strategies" section).
+
+- Ensuring the Signed-off-by-trailer
+  Contributors need to sign the "Contributor License Agreement". They do so by
+  signing off each individual commit with a so-called "Signed-off-by Trailer".
+  Git offers the `--signoff`/`-s` flags when using `git-commit` to commit
+  changes.
+
+  To ensure that all commits have the trailer present in the commit message, a
+  CI lint is installed in the github actions workflows of the project. This lint
+  blocks pull requests to be merged if one or more of the commits of the pull
+  request misses the `Signed-off-by` trailer.
+
+  As a result, it is not possible to merge pull requests that miss the trailer.
+- coding styleguide
+  Coding style is enforced via `rustfmt` in its default configuration.
+  Compliance with the coding style is enforced via CI.
+- testing
+  Testing is done via workflows in github actions using `cargo test` for all
+  crates.
+  The main objective with this is that a developer should be able to simply run
+  `cargo test` on their local machine to be able to see whether CI would succeed
+  for changes they submit in a pull request.
+  Thus, all CI tests (unit tests, but also integration tests) are implemented in
+  Rust and do not rely on external services.
+  End-to-end system tests - that depend on external services - are run outside the CI pipeline,
+  to avoid inconsistent test outcomes because of external issues.
+- benchmarks
+- documentation builds
+  Source code documentation as well as other documentation is tested via github
+  actions workflows as well, to ensure that a developer is able to build all
+  relevant documentation on their local machine.
+- keeping spec up to date
+- evergreen master
+  The project pursues the "evergreen master" strategy. That means that at every
+  point in time, the commit that `master`/`main` points to must successfully
+  build and pass all tests successfully.
+
+  Reaching that goal is possible because of the employed merge strategies (read
+  below).
+- merge strategies
+  Merging is the way how the project accepts and implements changes.
+
+  Because of the pursued "everygreen master", merging is implemented in a
+  certain way that forbids several bad practices which could lead to a breaking
+  build on the `master`/`main` branch, and a special workflow is implemented to
+  ensure not only the "evergreen master" but also to streamline the workflow for
+  integrating pull requests.
+
+  The following actions are **not** allowed:
+
+  * Committing to `master`/`main` directly
```

Comment:
How about tags?

Reply:
What would you tag directly on master?

Reply:
> As of now, for the patch version up (the last digit increment, e.g. 0.7.x), we tag the release number directly on main branch.

Ah, so that would mean that patch releases contained new features?

I guess we should re-think that, yes. If you want me to, I can come up with a proposal and we can discuss this (in another issue though)!

Reply:
Could you @matthiasbeyer together with @rina23q write a 'operation book' that summarizes all the individual steps needed to:

- deploy a new minor version
- deploy a new patch version ?

Including on how to backport, bump version numbers, tagging etc. So we can all agree on a flow and ensure that potentially multiple people could release in case of vacations, leaves, etc...

Reply:
Sure! :tada: - just to clearify: By "deploy" you mean "release", because... deployment should always be task of the Operator/Administrator of the infrastructure... we're just developing here... :smile: :question:

Reply:
Also: Where should this be written down? In the document from this PR or somewhere else? If somewhere else, where? And if somewhere else, it might be a good idea to open a dedicated issue for that.

Reply:
@TheNeikos Yes, it makes sense. We should write down somewhere clearly. I understand the main scope is "how to release minor/patch versions". And the doc needs to be include
- branching (maybe also code freeze?)
- how to backport
- bump version numbers
- tagging
- release note / change log
- could be more

Reply:
Yup exactly. Deploy here being the 'upload to crates.io process'.

Great!

Reply:
> Yup exactly.

I take that as "Yes, this should happen in this document, please open a PR towards this PR" :laughing:

Reply:
I consider this done.